packages/nightingale-sunburst/src/nightingale-sunburst.js

```diff
@@ -0,0 +1,358 @@
+import { LitElement, html } from "lit-element";
+import {
+  select,
+  partition as d3partition,
+  hierarchy,
+  scaleOrdinal,
+  quantize,
+  interpolateRainbow,
+} from "d3";
+
+const superkingdoms = ["bacteria", "viruses", "archaea", "eukaryota", null];
+
+const partition = (data, radius, attribute) => {
+  const h = hierarchy(data).sort((a, b) => b[attribute] - a[attribute]);
+  return d3partition().size([2 * Math.PI, radius])(h);
+};
+
+const getValue = (d, attributeName) => {
+  if (d[attributeName]) return d[attributeName];
+  if (!d.children) return 0;
+  return d.children.reduce(
+    (agg, item) => agg + getValue(item, attributeName),
+    0
+  );
+};
+
+const prepareTreeData = (node, depth, maxDepth, weightAttribute) => {
```

Comment:
Not a big fan of mutating the original data, in case it is shared with logic elsewhere.
I would keep the data in the component, untouched in _data, and prepare tree data would generate a new object, recursively cloned from the original data with all the required modifications done here, and assigned to a private _tree, _treeData, or _internalData, that would be used to draw.